File: src/test/java/com/flexpoker/table/command/aggregate/singlehand/twoplayer/TwoPlayerRaiseBySmallBlindAndBigBlindFoldsTest.java
package com.flexpoker.table.command.aggregate.singlehand.twoplayer;

import static com.flexpoker.test.util.CommonAssertions.verifyAppliedAndNewEventsForAggregate;

import java.util.UUID;

import org.junit.Test;

import com.flexpoker.table.command.aggregate.testhelpers.TableTestUtils;
import com.flexpoker.table.command.events.ActionOnChangedEvent;
import com.flexpoker.table.command.events.CardsShuffledEvent;
import com.flexpoker.table.command.events.HandCompletedEvent;
import com.flexpoker.table.command.events.HandDealtEvent;
import com.flexpoker.table.command.events.LastToActChangedEvent;
import com.flexpoker.table.command.events.PlayerFoldedEvent;
import com.flexpoker.table.command.events.PlayerRaisedEvent;
import com.flexpoker.table.command.events.PotAmountIncreasedEvent;
import com.flexpoker.table.command.events.PotCreatedEvent;
import com.flexpoker.table.command.events.RoundCompletedEvent;
import com.flexpoker.table.command.events.TableCreatedEvent;
import com.flexpoker.table.command.events.WinnersDeterminedEvent;

public class TwoPlayerRaiseBySmallBlindAndBigBlindFoldsTest {

    @Test
    public void test() {
        var tableId = UUID.randomUUID();

        var table = TableTestUtils.createBasicTableAndStartHand(tableId, UUID.randomUUID(), UUID.randomUUID());

        // use the info in action on event to determine who the small blind/button is on
        var smallBlindAndButtonPlayerId = ((ActionOnChangedEvent) table.fetchNewEvents().get(4)).getPlayerId();
        table.raise(smallBlindAndButtonPlayerId, 40);

        var bigBlindPlayerId = ((ActionOnChangedEvent) table.fetchNewEvents().get(6)).getPlayerId();
        table.fold(bigBlindPlayerId);

        verifyAppliedAndNewEventsForAggregate(table,
                TableCreatedEvent.class, CardsShuffledEvent.class,
                HandDealtEvent.class, PotCreatedEvent.class,
                ActionOnChangedEvent.class, PlayerRaisedEvent.class,
                ActionOnChangedEvent.class, LastToActChangedEvent.class,
                PlayerFoldedEvent.class, PotAmountIncreasedEvent.class,
                PotAmountIncreasedEvent.class, RoundCompletedEvent.class,
                WinnersDeterminedEvent.class, HandCompletedEvent.class);
    }

}

File: src/test/kotlin/com/flexpoker/table/command/aggregate/singlehand/twoplayer/TwoPlayerRaiseBySmallBlindAndBigBlindFoldsTest.kt
package com.flexpoker.table.command.aggregate.singlehand.twoplayer

import com.flexpoker.table.command.aggregate.applyEvents
import com.flexpoker.table.command.aggregate.eventproducers.fold
import com.flexpoker.table.command.aggregate.eventproducers.raise
import com.flexpoker.table.command.aggregate.testhelpers.blindPlayerIds
import com.flexpoker.table.command.aggregate.testhelpers.createBasicTableAndStartHand
import com.flexpoker.table.command.events.ActionOnChangedEvent
import com.flexpoker.table.command.events.CardsShuffledEvent
import com.flexpoker.table.command.events.HandCompletedEvent
import com.flexpoker.table.command.events.HandDealtEvent
import com.flexpoker.table.command.events.LastToActChangedEvent
import com.flexpoker.table.command.events.PlayerFoldedEvent
import com.flexpoker.table.command.events.PlayerRaisedEvent
import com.flexpoker.table.command.events.PotAmountIncreasedEvent
import com.flexpoker.table.command.events.PotCreatedEvent
import com.flexpoker.table.command.events.RoundCompletedEvent
import com.flexpoker.table.command.events.TableCreatedEvent
import com.flexpoker.table.command.events.WinnersDeterminedEvent
import com.flexpoker.test.util.CommonAssertions.verifyNewEvents
import com.flexpoker.test.util.TableEventProducerApplierBuilder
import org.junit.jupiter.api.Test
import java.util.UUID

class TwoPlayerRaiseBySmallBlindAndBigBlindFoldsTest {

    @Test
    fun test() {
        val tableId = UUID.randomUUID()
        val events = createBasicTableAndStartHand(tableId, UUID.randomUUID(), UUID.randomUUID())
        val initState = applyEvents(events)

        val (buttonOnPlayerId, smallBlindPlayerId, bigBlindPlayerId) = blindPlayerIds(initState)

        val (_, newEvents) = TableEventProducerApplierBuilder()
            .initState(initState)
            .andRun { raise(it, smallBlindPlayerId, 40) }
            .andRun { fold(it, bigBlindPlayerId) }
            .run()

        verifyNewEvents(tableId, events + newEvents,
            TableCreatedEvent::class.java, CardsShuffledEvent::class.java,
            HandDealtEvent::class.java, PotCreatedEvent::class.java,
            ActionOnChangedEvent::class.java, PlayerRaisedEvent::class.java,
            ActionOnChangedEvent::class.java, LastToActChangedEvent::class.java,
            PlayerFoldedEvent::class.java, PotAmountIncreasedEvent::class.java,
            PotAmountIncreasedEvent::class.java, RoundCompletedEvent::class.java,
            WinnersDeterminedEvent::class.java, HandCompletedEvent::class.java
        )
    }

}